packages/data/README.md

```diff
@@ -793,9 +793,25 @@ any price in the state for that currency is retrieved. If the currency prop
 doesn't change and other props are passed in that do change, the price will
 not change because the dependency is just the currency.
 
+When data is only used in an event callback, the data should not be retrieved
+on render, so it may be useful to get the selectors function instead.
+
+```js
+import { useSelect } from '@wordpress/data';
+
+function Paste( { children } ) {
+  const { getSettings } = useSelect( 'my-shop' );
+  function onPaste() {
+    // Do something with the settings.
+    const settings = getSettings();
+  }
+  return <div onPaste={ onPaste }>{ children }</div>;
+}
+```
+
 _Parameters_
 
--   _\_mapSelect_ `Function`: Function called on every state change. The returned value is exposed to the component implementing this hook. The function receives the `registry.select` method on the first argument and the `registry` on the second argument.
+-   _\_mapSelect_ `Function|WPDataStore|string`: Function called on every state change. The returned value is exposed to the component implementing this hook. The function receives the `registry.select` method on the first argument and the `registry` on the second argument. When a store key is passed, all selectors for the store will be returned. This is only meant for usage of these selectors in event callbacks, not for data needed to create the element tree.
```

Comment:
I added documentation now. I feel like we're overloading the `useSelect` a bit though, so this may be another argument to create a separate `useSelectForCallbacks` hook. It could be simpler code-wise as well since we don't need to add logic for skipping the mapping.

Reply:
Either way, it would be rather confusing whatever we do at this stage. We could have `useSelectors` or something like that but you still need to read docs to understand the difference so maybe it's fine to add jQuery-like flexibility in terms what passed arguments trigger 😄